src/components/cuda/cupti_common.c

```diff
@@ -0,0 +1,465 @@
+{
+    if (dl_rt) {
+        dlclose(dl_rt);
+        dl_rt = NULL;
+    }
+    cudaGetDevicePtr           = NULL;
+    cudaGetDeviceCountPtr      = NULL;
+    cudaGetDevicePropertiesPtr = NULL;
+    cudaDeviceGetAttributePtr  = NULL;
+    cudaSetDevicePtr           = NULL;
+    cudaFreePtr                = NULL;
+    cudaDriverGetVersionPtr    = NULL;
+    cudaRuntimeGetVersionPtr   = NULL;
+    return PAPI_OK;
+}
+
+static int load_cupti_common_sym(void)
+{
+    char dlname[] = "libcupti.so";
+    char lookup_path[PATH_MAX];
+
+    char *papi_cuda_cupti = getenv("PAPI_CUDA_CUPTI");
+    if (papi_cuda_cupti) {
+        sprintf(lookup_path, "%s/%s", papi_cuda_cupti, dlname);
+        dl_cupti = dlopen(lookup_path, RTLD_NOW | RTLD_GLOBAL);
+    }
+
+    const char *standard_paths[] = {
+        "%s/extras/CUPTI/lib64/%s",
+        "%s/lib64/%s",
+        NULL,
+    };
+
+    char *papi_cuda_root = getenv("PAPI_CUDA_ROOT");
+    if (papi_cuda_root && !dl_cupti) {
+        dl_cupti = load_dynamic_syms(papi_cuda_root, dlname, standard_paths);
+    }
+
+    if (!dl_cupti) {
+        dl_cupti = dlopen(dlname, RTLD_NOW | RTLD_GLOBAL);
+        if (!dl_cupti) {
+            ERRDBG("Loading libcupti.so failed. Try setting PAPI_CUDA_ROOT\n");
+            goto fn_fail;
+        }
+    }
+
+    cuptiGetVersionPtr = DLSYM_AND_CHECK(dl_cupti, "cuptiGetVersion");
+
+    Dl_info info;
+    dladdr(cuptiGetVersionPtr, &info);
+    LOGDBG("CUPTI library loaded from %s\n", info.dli_fname);
+    return PAPI_OK;
+fn_fail:
+    return PAPI_EMISC;
+}
+
+static int unload_cupti_common_sym(void)
+{
+    if (dl_cupti) {
+        dlclose(dl_cupti);
+        dl_cupti = NULL;
+    }
+    cuptiGetVersionPtr = NULL;
+    return PAPI_OK;
+}
+
+static int util_load_cuda_sym(void)
+{
+    int papi_errno;
+    papi_errno = load_cuda_sym();
+    papi_errno += load_cudart_sym();
+    papi_errno += load_cupti_common_sym();
+    if (papi_errno != PAPI_OK) {
+        return PAPI_EMISC;
+    }
+    else
+        return PAPI_OK;
+}
+
+int util_unload_cuda_sym(void)
+{
+    unload_cuda_sym();
+    unload_cudart_sym();
+    unload_cupti_common_sym();
+    return PAPI_OK;
+}
+
+static int util_dylib_cu_runtime_version(void)
+{
+    int runtimeVersion;
+    CUDART_CALL(cudaRuntimeGetVersionPtr(&runtimeVersion), return PAPI_EMISC );
+    return runtimeVersion;
+}
+
+static int util_dylib_cupti_version(void)
+{
+    unsigned int cuptiVersion;
+    CUPTI_CALL(cuptiGetVersionPtr(&cuptiVersion), return PAPI_EMISC );
+    return cuptiVersion;
+}
+
+int get_device_count(void)
+{
+    static int numDevs = -1;
+    if (numDevs != -1)
+        goto fn_exit;
+    CUDART_CALL(cudaGetDeviceCountPtr(&numDevs), return PAPI_EMISC);
+fn_exit:
+    return numDevs;
+}
+
+static int get_gpu_compute_capability(int dev_num)
+{
+    int cc_major, cc_minor;
+    int cc;
+    CUDART_CALL(cudaDeviceGetAttributePtr(&cc_major,
+        cudaDevAttrComputeCapabilityMajor, dev_num),
+            return PAPI_EMISC );
```

Comment:
either have a white space after `(` and before `)` or no white space at all. Don't mix things up.

Reply:
Do you mean follow the same convention throughout, or on a per statement basis?

Reply:
throughout possibly. Unfortunately we don't enforce coding standards but we should do it eventually. Too much inconsistency across the code. So I am not picking about the specific style just pick one and stick with it. We can fix the style eventually to be consistent across the code base.